pkgs/tools/misc/hdf4/default.nix

```diff
@@ -31,11 +31,12 @@ stdenv.mkDerivation rec {
     "-DHDF4_BUILD_WITH_INSTALL_NAME=OFF"
     "-DHDF4_ENABLE_JPEG_LIB_SUPPORT=ON"
     "-DHDF4_ENABLE_NETCDF=OFF"
-    "-DHDF4_ENABLE_SZIP_ENCODING=ON"
-    "-DHDF4_ENABLE_SZIP_SUPPORT=ON"
     "-DHDF4_ENABLE_Z_LIB_SUPPORT=ON"
     "-DHDF4_BUILD_FORTRAN=OFF"
     "-DJPEG_DIR=${libjpeg}"
+  ] ++ stdenv.lib.optionals (szip != null) [
+    "-DHDF4_ENABLE_SZIP_ENCODING=ON"
```

Comment:
> Now, there is another class of users, the commercial user with a license to use szip encoding (or someone who believes the patent is not valid (which, incidentally is everyone in Europe(!)) who could still build the software including all the features, in particular the szip encoding (which is "-DHDF4_ENABLE_SZIP_ENCODING=ON").

It is false that for everyone in Europe () building `szip` is legally safe, as the restriction is in copyright license, too.

> if the user is building from source, then ENCODING should be ON.

It is false becaue even for local builds NixPkgs should produce free software unless explicitly specified, and everyone agrees that `szip` is unfree and so linking against it should be avoided.

> other PRs (with higher quality code)

And the ease of review is usually related to the number of lines changed, I don't know how to make GitHub sort by that… And this is first-time contributor.

As for the quality of code, I consider this way to make this change to be one of the optimal ways.